src/systemFlake.nix

```diff
@@ -189,24 +192,37 @@ mergeAny otherArguments (
 
         pkgs = mapAttrs importChannel channels;
 
-        mkOutput = output: builder:
+
+        deprecatedBuilders = channels: { }
+        // optionalAttrs (packagesBuilder != null) { packages = packagesBuilder channels; }
+        // optionalAttrs (defaultPackageBuilder != null) { defaultPackage = defaultPackageBuilder channels; }
+        // optionalAttrs (appsBuilder != null) { apps = appsBuilder channels; }
+        // optionalAttrs (defaultAppBuilder != null) { defaultApp = defaultAppBuilder channels; }
+        // optionalAttrs (devShellBuilder != null) { devShell = devShellBuilder channels; }
+        // optionalAttrs (checksBuilder != null) { checks = checksBuilder channels; };
+
+        systemOutputs = (if outputsBuilder == null then deprecatedBuilders else outputsBuilder) pkgs;
+
+        mkOutput = output:
           mergeAny
             # prevent override of nested outputs in otherArguments
             (optionalAttrs (otherArguments ? ${output}.${system})
               { ${output} = otherArguments.${output}.${system}; })
-            (optionalAttrs (args ? ${builder})
-              { ${output} = args.${builder} pkgs; });
+            (optionalAttrs (systemOutputs ? ${output})
+              { ${output} = systemOutputs.${output}; });
+
```

Comment:
I think this would be better done with a `mapAttrs`. That way its future proofed if more system-spaced outputs ever get added.

Reply:
also code is a bit simpler, `mkOutput` is no longer necessary

Reply:
Yes, I agree mapAttrs would be better option but we are dealing with multiple restrictions:
- `nix flake show` should not error - we cant output empty attrsets
- outputs must be merged - if we just go with mapAttrs while keeping in mind the first point - not all attrs will get merged.

And yeah, I guess I could go over the names and just generate attrset out of that. Probably will change that but I really should go and get some sleep now :D

Reply:
sounds good, I think it could probably be used in the future once we remove the deprecated arguments. For now this works well.

Reply:
Not going to do the attrs from list thingy. At least for time being - could not find a solution that I'd be happy with. Its other solutions are more complex and use around the same amount of lines